GitHub_LearningTests/Generic/Utilities/Infrastructure/StringExtensionTests.swift

```diff
@@ -0,0 +1,154 @@
+//
+//  StringExtensionTests.swift
+//  GitHub_LearningTests
+//
+//  Created by Amin Ashhab on 16.12.2020.
+//  Copyright © 2020 Amin_Second_Test_Project. All rights reserved.
+//
+
+import XCTest
+@testable import GitHub_Learning
+
+class StringExtensionTests: XCTestCase {
+    private var sut: String!
+    
+    func setUpTestWith(string: String) {
+        sut = string
+    }
+    
+    func test_givenNumberInString_thenStringContainsNumber() {
+        // Given
+        setUpTestWith(string: "1")
+        
+        // Then
+        XCTAssertTrue(sut.containsNumber)
+    }
+    
+    func test_givenNoNumberInString_thenStringDoNotContainsNumber() {
+        // Given
+        setUpTestWith(string: "a")
+        
+        // Then
+        XCTAssertFalse(sut.containsNumber)
+    }
+    
+    func test_givenCapitalInString_thenStringContainsCapital() {
+        // Given
+        setUpTestWith(string: "A")
+        
+        // Then
+        XCTAssertTrue(sut.containsCapital)
+    }
+    
+    func test_givenNoCapitalInString_thenStringDoNotContainsCapital() {
+        // Given
+        setUpTestWith(string: "a")
+        
+        // Then
+        XCTAssertFalse(sut.containsCapital)
+    }
+    
+    func test_givenTwoLettersInString_thenStringsLenghtIsTwo() {
+        // Given
+        setUpTestWith(string: "ab")
+        
+        // Then
+        XCTAssertTrue(sut.length == 2)
+    }
+    
+    func test_givenOneLettersInString_thenStringsLenghtIsNotTwo() {
+        // Given
+        setUpTestWith(string: "a")
+        
+        // Then
+        XCTAssertFalse(sut.length == 2)
```

Comment:
Delku testovat nemusis. Co chces testovat jsou ty tvoje metody.

Reply:
length je můj custom computed var.

Reply:
muzes pouzit `count`. Dela uplne to same.

Reply:
To vim jen jsem si to chtěl přejmenovat aby to bylo čtivější. Tak to nemám testovat teda?